### PR TITLE
[Fix](Nereids)add comments to CostAndEnforcerJob and fix view test case

### PR DESCRIPTION
**Pull request by sohardforaname**
# Proposed changes

Issue Number: close #xxx

## Problem summary

1. add comments to cost and enforcer job as some code is too hard to understand
2. fix nereids_syntax_p0/view.groovy's multi-answer bug.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
3. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
4. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
5. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
6. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221014164452_clickbench_pr_29409.html

**Comment by hello-stephen** [comment]
https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221015000153_clickbench_pr_29615.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 41.14 seconds
 load time: 635 seconds
 storage size: 17154810775 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221021104935_clickbench_pr_32505.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 39.05 seconds
 load time: 562 seconds
 storage size: 17154655327 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221021163947_clickbench_pr_32605.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 39.35 seconds
 load time: 581 seconds
 storage size: 17154821303 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221024140205_clickbench_pr_33183.html

File: build.sh [hunk 1]
```diff
@@ -433,9 +433,9 @@ function build_ui() {
 }
 
 # FE UI must be built before building FE
-if [[ "${BUILD_FE}" -eq 1 ]]; then
-    build_ui
-fi
+# if [[ "${BUILD_FE}" -eq 1 ]]; then
+#     build_ui
```
Comment (jackwener) [inline build.sh:437]: remove this

File: fe/fe-core/src/main/java/org/apache/doris/nereids/NereidsPlanner.java [hunk 2]
```diff
@@ -73,7 +75,22 @@ public void plan(StatementBase queryStmt, org.apache.doris.thrift.TQueryOptions
...
-        PhysicalPlan physicalPlan = plan(logicalPlanAdapter.getLogicalPlan(), PhysicalProperties.ANY);
+        PhysicalPlan physicalPlan = null;
+        try {
+            physicalPlan = plan(logicalPlanAdapter.getLogicalPlan(), PhysicalProperties.ANY);
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new AnalysisException(e.getMessage());
+        } finally {
+            try {
+                FileOutputStream fs = new FileOutputStream("/mnt/disk1/mch/projects/doris/fe.log");
```
Comment (jackwener) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/NereidsPlanner.java:86]: Shouldn't add.

File: build.sh (hunk 1 above)
Comment (sohardforaname) [reply build.sh:437]: ok

File: fe/fe-core/src/main/java/org/apache/doris/nereids/NereidsPlanner.java (hunk 2 above)
Comment (sohardforaname) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/NereidsPlanner.java:86]: ok

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 38.48 seconds
 load time: 564 seconds
 storage size: 17154644837 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221027092103_clickbench_pr_34715.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 38.2 seconds
 load time: 563 seconds
 storage size: 17154644860 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221028050041_clickbench_pr_35135.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 39.03 seconds
 load time: 576 seconds
 storage size: 17154644801 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221028104337_clickbench_pr_35327.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 38.67 seconds
 load time: 558 seconds
 storage size: 17154655392 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221028123446_clickbench_pr_35401.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 39.38 seconds
 load time: 562 seconds
 storage size: 17154655385 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221028124316_clickbench_pr_35408.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 39.24 seconds
 load time: 565 seconds
 storage size: 17154655409 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221028125845_clickbench_pr_35419.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 39.57 seconds
 load time: 565 seconds
 storage size: 17154644786 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221028130542_clickbench_pr_35425.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 38.87 seconds
 load time: 565 seconds
 storage size: 17154705986 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221028132140_clickbench_pr_35443.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 38.41 seconds
 load time: 558 seconds
 storage size: 17154655320 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221028134126_clickbench_pr_35457.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 38.27 seconds
 load time: 560 seconds
 storage size: 17154644825 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221028141257_clickbench_pr_35479.html

File: fe/fe-core/src/main/java/org/apache/doris/nereids/memo/GroupExpression.java [hunk 3]
```diff
@@ -168,25 +182,24 @@ public List<PhysicalProperties> getInputPropertiesList(PhysicalProperties requir
...
             if (lowestCostTable.get(outputProperties).first > cost) {
                 lowestCostTable.put(outputProperties, Pair.of(cost, childrenInputProperties));
                 return true;
-            } else {
-                return false;
             }
-        } else {
-            lowestCostTable.put(outputProperties, Pair.of(cost, childrenInputProperties));
-            return true;
+            return false;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/memo/GroupExpression.java:195]: do not remove else block. it is easy to do a accidentally modifying with out it

File: fe/fe-core/src/main/java/org/apache/doris/nereids/memo/GroupExpression.java [hunk 4]
```diff
@@ -109,6 +114,14 @@ public void setChildren(ImmutableList<Group> children) {
...
     }
 
+    public boolean isHasCalculateCost() {
+        return hasCalculateCost;
+    }
+
+    public void setHasCalculateCost(boolean hasCalculateCost) {
+        this.hasCalculateCost = hasCalculateCost;
+    }
+
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/memo/GroupExpression.java:124]: remove it in this PR

File: fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/cascades/CostAndEnforcerJob.java [hunk 5]
```diff
@@ -164,74 +167,96 @@ public void execute() {
...
                 if (curTotalCost < context.getCostUpperBound()) {
                     context.setCostUpperBound(curTotalCost);
                 }
             }
-
             clear();
         }
     }
 
+    private boolean calculateEnforce(List<PhysicalProperties> requestChildrenProperties) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/cascades/CostAndEnforcerJob.java:189]: add some comment to explain return value

File: fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/cascades/CostAndEnforcerJob.java [hunk 6]
```diff
@@ -164,74 +167,100 @@ public void execute() {
...
+
+        // enforcedProperty is superset of requiredProperty
+        if (!addEnforcedProperty.equals(requiredProperties)) {
+            recordPropertyAndCost(groupExpression.getOwnerGroup().getBestPlan(addEnforcedProperty),
+                    addEnforcedProperty, requiredProperties, Lists.newArrayList(outputProperty));
         }
     }
 
+    /**
+     * record property and cost
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/cascades/CostAndEnforcerJob.java:258]: should add a blank line between method doc and param docs. plz fix it in next PR

**Review by morrySnow — APPROVED** [review]

**Comment by morrySnow** [comment]
1. add comments to cost and enforcer job as some code is too hard to understand
2. fix nereids_syntax_p0/view.groovy's multi-answer bug.

**Pull request merged**